microsoft-edge/webview2/how-to/debug.md

```diff
@@ -101,148 +110,169 @@ To debug your WebView2 app:
 
+#### Virtual source path mapping not supported in Visual Studio 2019
+
+If you use the WebView2 [SetVirtualHostNameToFolderMapping](/dotnet/api/microsoft.web.webview2.core.corewebview2.setvirtualhostnametofoldermapping) method, the debugger in Visual Studio 2019 doesn't understand the virtual source path mapping, so breakpoints don't work correctly.
 
+Virtual source path mapping works when you use the debugger in Visual Studio Code.<!-- or in Visual Studio later than 2019. -->
+
+
+<!-- might need to keep blank lines around tab headers -->
 <!-- ====================================================================== -->
 
 ## [Visual Studio Code](#tab/visualstudiocode)
 
-Use Microsoft Visual Studio Code to debug scripts that run in WebView2 controls.  <!--Ensure that you're using Visual Studio Code version [insert build here] or later.  -->
+Use Microsoft Visual Studio Code to debug scripts that run in WebView2 controls.  <!-- Make sure you're using Visual Studio Code version [insert build here] or later. -->
 
-In Visual Studio Code, complete the following actions to debug your code.
 
-1. Your project is required to have a `launch.json` file.  If your project doesn't have a `launch.json` file, create a new `launch.json` file and copy the following code into it:
+### Create a launch.json file
 
-    ```json
-        "name": "Hello debug world",
-        "type": "pwa-msedge",
-        "port": 9222, // The port value is optional, and the default value is 9222.
-        "request": "launch",
-        "runtimeExecutable": "C:/path/to/your/webview2/app.exe",
-        "env": {
-            // Customize for your app location if needed
-            "Path": "%path%;e:/path/to/your/app/location; "
-        },
-        "useWebView": true,
-        // The following two lines setup source path mapping, where `url` is the start page of your app, and `webRoot` is the top level directory with all your code files.
-        "url": "file:///${workspaceFolder}/path/to/your/toplevel/foo.html",
-        "webRoot": "${workspaceFolder}/path/to/your/assets"
-    ```
+1. To debug your code, your project is required to have a `launch.json` file.  If your project doesn't have a `launch.json` file, create a new `launch.json` file and paste the following code into it:
 
-    > [!NOTE]
-    > Visual Studio Code source path mapping now requires the URL, so your app now receives a command-line parameter when it starts.  You can safely ignore the `url` parameter, if needed.
+   ```json
+      "name": "Hello debug world",
+      "type": "pwa-msedge",
+      "port": 9222, // The port value is optional, and the default value is 9222.
```

Comment:
Convert // comments in .json listings to .json objects?
"_comment": "comment text goes here...",